etl/steps/data/garden/tuberculosis/2023-11-27/burden_disaggregated.py

```diff
@@ -0,0 +1,183 @@
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def combining_sexes_for_all_age_groups(tb: Table) -> Table:
+    """
+    Not all of the age-groups provided by the WHO have a value for both sexes, so we need to combine values for males and females to calculate these.
+    """
+
+    tb["age_group"] = tb["age_group"].astype("str")
+    age_groups_with_both_sexes = tb[tb["sex"] == "a"]["age_group"].drop_duplicates().to_list()
+    msk = tb["age_group"].isin(age_groups_with_both_sexes)
+    tb_age = tb[~msk]
+    tb_gr = (
+        tb_age.groupby(["country", "year", "age_group", "risk_factor"], dropna=False)[
+            ["best", "lo", "hi", "population"]
+        ]
+        .sum()
+        .reset_index()
+    )
+    tb_gr["sex"] = "a"
+    # Set population to nan for rows where the risk factor is not "all"
+    tb_gr.loc[tb_gr["risk_factor"] != "all", "population"] = np.nan
+    tb = pr.concat([tb, tb_gr], axis=0, ignore_index=True, short_name=paths.short_name)
+
+    return tb
+
+
+def add_region_sum_aggregates(tb: Table, ds_regions: Dataset, ds_income_groups: Dataset) -> Table:
+    """
+    Calculate region aggregates for all for each combination of age-group, sex and risk factor in the dataset.
+    """
+    # Create the groups we want to aggregate over.
+    tb_gr = tb.groupby(["year", "age_group", "sex", "risk_factor"])
+    tb_gr_out = Table()
+    for gr_name, gr in tb_gr:
+        for region in REGIONS_TO_ADD:
+            # List of countries in region.
+            countries_in_region = geo.list_members_of_region(
+                region=region,
+                ds_regions=ds_regions,
+                ds_income_groups=ds_income_groups,
+            )
+            gr_cal = gr[["country", "year", "best", "lo", "hi", "population"]]
+            # Add region aggregates.
+            gr_reg = geo.add_region_aggregates(
+                df=gr_cal,
+                region=region,
+                countries_in_region=countries_in_region,
+                frac_allowed_nans_per_year=0.5,
+                num_allowed_nans_per_year=None,
+            )
+            # Take only region values
+            gr_reg = gr_reg[gr_reg["country"] == region]
+            # Ensure the region values are assigned the same group values as the original group.
+            gr_reg[["age_group", "sex", "risk_factor"]] = [gr_name[1], gr_name[2], gr_name[3]]
+            # Combine the region values with the original group.
+            gr = pr.concat([gr, gr_reg], axis=0, ignore_index=True, copy=False)
+        # Add the group to the output table.
+        tb_gr_out = pr.concat([tb_gr_out, gr], axis=0, ignore_index=True, short_name=paths.short_name)
+
+    return tb_gr_out
+
+
+def add_population_column(tb: Table) -> Table:
+    """
+    Adding the population for each age-group, in rows where the risk factor is "all".
+    """
+    tb_pop = tb[tb["risk_factor"] == "all"]
+    tb_no_pop = tb[tb["risk_factor"] != "all"]
```

Comment:
To avoid warnings, add `reset_index(drop=True)` after slicing dataframes/tables.